avaframe/out3Plot/outQuickPlot.py

```diff
@@ -27,22 +25,116 @@
 log = logging.getLogger(__name__)
 
 
+def generatePlot(dataDict, avaName, outDir, cfg):
+    """ Create plots of two ascii datasets that shall be compared  """
+
+    # Extract info for plotting
+    data1 = dataDict['data1']
+    data2 = dataDict['data2']
+    name1 = dataDict['name1']
+    name2 = dataDict['name2']
+    cellSize = dataDict['cellSize']
+    if dataDict['compareType'] == 'compToRef':
+        simName = dataDict['simName'] + '_' + dataDict['suffix']
+        unit = dataDict['unit']
+    else:
+        simName = 'compare'
+        unit = ''
+
+    # Set dimensions of plots
+    ny = data1.shape[0]
+    nx = data1.shape[1]
+    Ly = ny*cellSize
+    Lx = nx*cellSize
+
+    # Location of Profiles
+    ny_loc = int(nx * 0.5)
+    nx_loc = int(ny * 0.5)
+
+    # Difference between datasets
+    dataDiff = data1 - data2
+    diffMax = np.amax(dataDiff)
+    diffMin = np.amin(dataDiff)
+    diffMean = np.mean(dataDiff)
+
+    # Location of box
+    nybox = int(nx * 0.2)
+    nxbox = int(ny * 0.2)
+
+    # Plot data
+    # Figure 1 shows the result parameter data
+    fig = plt.figure(figsize=(figW*3, figH))
+    suptitle = fig.suptitle(avaName, fontsize=14, color ='0.5')
+    ax1 = fig.add_subplot(131)
+    cmap = cmapGB
+    im1 = plt.imshow(data1, cmap=cmap, extent=[0, Lx, 0, Ly], origin='lower', aspect=nx/ny)
+    fig.colorbar(im1, ax=ax1)
+    ax1.set_aspect('auto')
+    title = str('%s' % name1)
+    ax1.set_title(title)
+    ax1.set_xlabel('x [m]')
+    ax1.set_ylabel('y [m]')
+    ax2 = fig.add_subplot(132)
+    im2 = plt.imshow(data2, cmap=cmap, extent=[0, Lx, 0, Ly], origin='lower', aspect=nx/ny)
+    fig.colorbar(im2, ax=ax2)
+    ax2.set_aspect('auto')
+    ax2.set_xlabel('x [m]')
+    title = str('%s' % name2)
+    ax2.set_title(title)
+    ax3 = fig.add_subplot(133)
+    cmap = cmapdiv
+    im3 = plt.imshow(dataDiff, cmap=cmap,
+                     extent=[0, Lx, 0, Ly], origin='lower', aspect=nx/ny)
+    fig.colorbar(im3, ax=ax3)
+    ax3.text(nybox, nxbox, 'Mean: %.2f %s\n Max: %.2f %s\n Min: %.2f %s' %
+            (diffMean, unit, diffMax, unit, diffMin, unit), bbox=dict(boxstyle="square", ec='white', fc='white'),
+            horizontalalignment='left', verticalalignment='bottom')
+    ax3.set_aspect('auto')
+    ax3.set_xlabel('x [m]')
+    ax3.set_title('Difference ref-sim')
+    fig.savefig(os.path.join(outDir, 'Diff_%s_%s.%s' % (avaName, simName, outputFormat)))#, bbox_extra_artists=(suptitle,), bbox_inches="tight")
```

Comment:
I think the bbox_extra is needed, right?

Reply:
Done